train.py

```diff
@@ -49,6 +49,27 @@ def ddp_main(rank, world_size, cfg):
         # restore the print function
         restore_print_override(original_print)
 
+def single_gpu_main(cfg):
```

Comment:
should keep this stuff or not?

Reply:
hmm, this seems duplicative and suggestive that our DDP isn't reliable. That's coz DDP should work even when GPU availability is 1. It'll work as a fail safe, but I think users will be doubtful if we've implemented other things reliably.

Reply:
So, I'm in favour not to keep it. Happy to hear @LeonGuertler 's thoughts